app/models/connected_app.rb

```diff
@@ -15,4 +16,16 @@ def connecting?
   def ready?
     !connecting?
   end
+
+  def connect(api_key:, channel:)
+    ConnectAppJob.perform_later(self, api_key, channel:)
+  end
+
+  def disconnect
+    WebhookDeliveryJob.perform_later(
+      data["destroy"],
+      "disconnect-app",
+      nil
+    )
+  end
```

Comment:
Yeah it feels wrong sending the websocket channel to the model. But with a service, I guess the controller has to know about the subtypes, and make the decision what to do with them, which doesn't seem any better.
 I'm curious what Maikel thinks too.

Reply:
I mean you could have a service that knows how to connect/disconnect each type of connected app, but that would add some indirection. 
It could make sense if the connect/disconnect was a setting not stored with the model.